### PR TITLE
Marginalizing gamma_in distributions of individual systems

### PR DESCRIPTION
**Pull request by williyamshoe**
This PR: 
- adds a feature to marginalize `gamma_in` distributions of individual systems at a population level.  
- adds @ajshajib and @williyamshoe as contributors in the AUTHORS.rst.

### CODE REVIEW

**Comment by williyamshoe** [comment]
From the code report, the tests fail because "Support for nose tests is deprecated".  Likely due to version differences with current version of pytest (v8.0.0).  Locally, my tests run fine (v7.4.3).  I am unsure what the solution is.  We can rewrite all our tests to conform to the new pytest version, or just use an older version.

**Comment by sibirrer** [comment]
> From the code report, the tests fail because "Support for nose tests is deprecated". Likely due to version differences with current version of pytest (v8.0.0). Locally, my tests run fine (v7.4.3). I am unsure what the solution is. We can rewrite all our tests to conform to the new pytest version, or just use an older version.

I just pushed a change hopefully making the deprecations/failures go away. Hope this works if you re-sync the main branch into your PR

File: hierarc/LensPosterior/kin_constraints_composite.py [hunk 1]
```diff
@@ -78,6 +80,8 @@ def __init__(
             line-of-sight velocity dispersion
         :param anisotropy_model: type of stellar anisotropy model. See details in
             MamonLokasAnisotropy() class of lenstronomy.GalKin.anisotropy
+        :param gamma_in_prior_mean: inner power-law slope of the NFW profile, if available
+        :param gamma_in_prior_std: 1-sigma error gamma_in
```
Comment (ajshajib) [inline hierarc/LensPosterior/kin_constraints_composite.py:84]:
```suggestion
        :param gamma_in_prior_std: prior standard deviation for `gamma_in`
```

File: hierarc/Likelihood/hierarchy_likelihood.py [hunk 2]
```diff
@@ -64,6 +66,8 @@ def __init__(
         :param normalized: bool, if True, returns the normalized likelihood, if False, separates the constant prefactor
          (in case of a Gaussian 1/(sigma sqrt(2 pi)) ) to compute the reduced chi2 statistics
         :param kwargs_lens_properties: keyword arguments of the lens properties
+        :param gamma_in_prior_mean: inner power-law slope of the NFW profile, if available
```
Comment (ajshajib) [inline hierarc/Likelihood/hierarchy_likelihood.py:69]:
```suggestion
        :param gamma_in_prior_mean: prior mean for inner power-law slope of the NFW profile, if available
```

File: hierarc/Likelihood/hierarchy_likelihood.py [hunk 3]
```diff
@@ -64,6 +66,8 @@ def __init__(
...
+        :param gamma_in_prior_std: 1-sigma error gamma_in
```
Comment (ajshajib) [inline hierarc/Likelihood/hierarchy_likelihood.py:70]:
```suggestion
        :param gamma_in_prior_std: 1-sigma error gamma_in
```

File: hierarc/Likelihood/cosmo_likelihood.py [hunk 4]
```diff
@@ -81,8 +81,8 @@ def __init__(
         :param anisotropy_distribution: string, distribution of the anisotropy parameters
         :param gamma_in_sampling: bool, if True samples gNFW inner slope parameter
         :param gamma_in_distribution: string, distribution function of the gamma_in parameter
-        :param m2l_sampling: bool, if True samples a global mass-to-light ratio parameter
-        :param m2l_distribution: string, distribution function of the m2l parameter
+        :param log_m2l_sampling: bool, if True samples a global mass-to-light ratio parameter
```
Comment (ajshajib) [inline hierarc/Likelihood/cosmo_likelihood.py:84]:
```suggestion
        :param log_m2l_sampling: bool, if True samples a global mass-to-light ratio parameter in log
```

File: hierarc/Sampling/ParamManager/lens_param.py [hunk 5]
```diff
@@ -33,9 +33,9 @@ def __init__(
         :param gamma_in_sampling: bool, if True samples the inner slope of the GNFW profile
         :param gamma_in_distribution: string, distribution function of the inner
             slope of the GNFW profile
-        :param m2l_sampling: bool, if True samples the mass to light ratio of
+        :param log_m2l_sampling: bool, if True samples the mass to light ratio of
             the stars
```
Comment (ajshajib) [inline hierarc/Sampling/ParamManager/lens_param.py:37]:
```suggestion
            the stars in logarithmic scale
```

File: hierarc/Sampling/ParamManager/lens_param.py [hunk 6]
```diff
@@ -33,9 +33,9 @@ def __init__(
...
-        :param m2l_distribution: string, distribution function of the mass to
+        :param log_m2l_distribution: string, distribution function of the mass to
```
Comment (ajshajib) [inline hierarc/Sampling/ParamManager/lens_param.py:38]:
```suggestion
        :param log_m2l_distribution: string, distribution function of the logarithm of mass to
```

File: hierarc/Sampling/ParamManager/param_manager.py [hunk 7]
```diff
@@ -64,11 +64,11 @@ def __init__(
         :param anisotropy_distribution: string, indicating the distribution function of the anisotropy model
         :param gamma_in_sampling: bool, if True samples gNFW inner slope parameter
         :param gamma_in_distribution: string, distribution function of the gamma_in parameter
-        :param m2l_sampling: bool, if True samples the mass-to-light ratio of the stars
-        :param m2l_distribution: string, distribution function of the m2l parameter
+        :param log_m2l_sampling: bool, if True samples the mass-to-light ratio of the stars
```
Comment (ajshajib) [inline hierarc/Sampling/ParamManager/param_manager.py:67]:
```suggestion
        :param log_m2l_sampling: bool, if True samples the mass-to-light ratio of the stars in logarithmic scale
```

File: hierarc/LensPosterior/kin_constraints_composite.py [hunk 8]
```diff
@@ -78,6 +80,8 @@ def __init__(
             line-of-sight velocity dispersion
         :param anisotropy_model: type of stellar anisotropy model. See details in
             MamonLokasAnisotropy() class of lenstronomy.GalKin.anisotropy
+        :param gamma_in_prior_mean: prior mean for inner power-law slope of the NFW profile, if available
+        :param gamma_in_prior_std: prior standard deviation for `gamma_in`
```
Comment (ajshajib) [inline hierarc/LensPosterior/kin_constraints_composite.py:84]:
```suggestion
        :param gamma_in_prior_std: standard deviation of the Gaussian prior for `gamma_in`
```

**Review by ajshajib — APPROVED** [review]

**Review by ajshajib — APPROVED** [review]

**Pull request merged**